packages/rpc-graphql/src/schema/account.ts

```diff
@@ -191,87 +192,104 @@ export const accountTypeDefs = /* GraphQL */ `
-                return 'AccountBase64';
-            }
-            if (account.encoding === 'base64+zstd') {
-                return 'AccountBase64Zstd';
-            }
-            if (account.encoding === 'jsonParsed') {
-                if (account.programName === 'nonce') {
-                    return 'NonceAccount';
-                }
-                if (account.accountType === 'mint' && account.programName === 'spl-token') {
-                    return 'MintAccount';
-                }
-                if (account.accountType === 'account' && account.programName === 'spl-token') {
-                    return 'TokenAccount';
+function accountResolvers(resolveCustomAccountTypeExtension?: (obj: unknown) => string) {
+    // TODO: This may not be the best default setup.
+    const resolveCustomAccountType = resolveCustomAccountTypeExtension ?? (_account => 'AccountBase64');
+    return {
+        Account: {
+            __resolveType(account: { encoding: string; programName: string; accountType: string }) {
+                if (account.encoding === 'base58') {
+                    return 'AccountBase58';
                 }
-                if (account.programName === 'stake') {
-                    return 'StakeAccount';
+                if (account.encoding === 'base64') {
+                    return 'AccountBase64';
                 }
-                if (account.accountType === 'vote' && account.programName === 'vote') {
-                    return 'VoteAccount';
+                if (account.encoding === 'base64+zstd') {
+                    return 'AccountBase64Zstd';
                 }
-                if (account.accountType === 'lookupTable' && account.programName === 'address-lookup-table') {
-                    return 'LookupTableAccount';
+                if (account.encoding === 'jsonParsed') {
+                    if (account.programName === 'nonce') {
+                        return 'NonceAccount';
+                    }
+                    if (account.accountType === 'mint' && account.programName === 'spl-token') {
+                        return 'MintAccount';
+                    }
+                    if (account.accountType === 'account' && account.programName === 'spl-token') {
+                        return 'TokenAccount';
+                    }
+                    if (account.programName === 'stake') {
+                        return 'StakeAccount';
+                    }
+                    if (account.accountType === 'vote' && account.programName === 'vote') {
+                        return 'VoteAccount';
+                    }
+                    if (account.accountType === 'lookupTable' && account.programName === 'address-lookup-table') {
+                        return 'LookupTableAccount';
+                    }
+                    return resolveCustomAccountType(account);
                 }
-            }
-            return 'AccountBase64';
+                return 'AccountBase64';
+            },
```

Comment:
This whole `__resolveType(..)` block didn't change, I just added this injectable resolve function at the end of the conditional.